packages/metro/src/node-haste/DependencyGraph.js

```diff
@@ -223,6 +223,19 @@ class DependencyGraph extends EventEmitter {
     this._haste.end();
   }
 
+  /** Given a search context, return a list of file paths matching the query. */
+  resolveContext(
+    from: string,
+    context: {
+      /* Should search for files recursively. Optional, default `true` when `require.context` is used */
+      recursive: boolean,
+      /* Filename filter pattern for use in `require.context`. Optional, default `/^\.\/.*$/` (any file) when `require.context` is used */
+      filter: RegExp,
+    },
```

Comment:
Nits:

1. Make read only
2. Drop defaults and optionality which don't apply here.
 
```suggestion
    context: $ReadOnly<{
      /* Should search for files recursively. */
      recursive: boolean,
      /* Filter relative paths against a pattern. */
      filter: RegExp,
    }>,
```